### PR TITLE
841: S3 file upload

### LINKED ISSUE #841: Implement photo endpoint & s3 integration Part 1 (Post)
**Describe the task**
Implement photo endpoint & s3 integration (post). Dependent on #840.

**Acceptance Criteria**
- [x] photo can be uploaded to s3
- [x] reference to photo in s3 lands in InvasivesBC database

### PR DESCRIPTION
**Pull request by NickPhura**
### Update to openapi 3
- Update api-mobile from [openapi 2.0 + swagger-tools] to [openapi 3.0 + express-openapi]
  - @micheal-w-wells & @popkinj Let me know if/when you want me to walk you through the new api setup.  It is not super different, but some of the existing code pieces live in different spots now.

### S3 file upload
- add file upload to activity endpoint
- update npm packages
- add new s3 env vars
- update dockerfiles with new env vars

### Other
- add SQL template string support
- removed the unused mailer env vars from the api-mobile configs/dockerfiles

### CODE REVIEW

File: api-mobile/src/paths/activity.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                  type: 'boolean'
+                }
+              }
+            },
+            media: {
+              description: 'The keys for the uploaded files',
+              type: 'array',
+              items: {
+                type: 'object',
+                required: ['fileName', 'encodedFile'],
```
Comment (micheal-w-wells) [inline api-mobile/src/paths/activity.ts:100]: I'm not seeing where the filename lives once the data hits the table, is it only stored in s3?

File: api-mobile/test/activity-with-files.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  "activityTypeData": {
+    "pokiness": 17
+  },
+  "activitySubTypeData": {
+    "density": "hard",
+    "colour": "green"
+  },
+  "media": [
+    {
+      "fileName": "bunnyFile",
```
Comment (micheal-w-wells) [inline api-mobile/test/activity-with-files.json]: Do I even want to know?

**Review by micheal-w-wells — DISMISSED** [review]

**Comment by micheal-w-wells** [comment]
Did the swagger file not have to change at all?  That's the only mystery to me right at this second (didn't see it in changed files)

File: api-mobile/test/activity-with-files.json (hunk 2 above)
Comment (NickPhura) [reply api-mobile/test/activity-with-files.json]: Haha, I was using a random image of a bunny for testing.

I do notice a small error in this test string though, so Ill fix that

File: api-mobile/src/paths/activity.ts (hunk 1 above)
Comment (NickPhura) [reply api-mobile/src/paths/activity.ts:100]: Good point, I think I missed including that.

We can chat tomorrow (Friday) about how we want to store the media information.  Maybe another json blob column, that can hold any number of media keys and their meta?  Can chat tomorrow.

**Comment by NickPhura** [comment]
> 
> 
> Did the swagger file not have to change at all? That's the only mystery to me right at this second (didn't see it in changed files)

The old swagger file is gone, replaced by one that is mostly the same.  The biggest difference, excluding some 2.0->3.0 related changes, is that the `paths: {}` section is now empty, because all of the path schema lives alongside the controller code.  
 - All of the paths/controllers live in `./src/paths` and are a 1-to-1 matchup with the routes they serve. 
   - So `GET <host>/api/observation/plant` is handled by the `GET` function in  `./src/paths/observation/plant.ts`
   - And `GET <host>/api/observation/plant/{plantId}` is handled by the `GET` function in `./src/paths/observation/plant/{plantId}.ts`
   - etc
 - The new base schema is in `./src/openapi/api-doc.yaml`.

File: api-mobile/src/paths/activity.ts (hunk 1 above)
Comment (NickPhura) [reply api-mobile/src/paths/activity.ts:100]: Oh, it is in the raw payload we store as well. 

The media array that gets sent in the request `[ { fileName: string, encodedFile: string } ]` gets saved, just with the `encodedFile` property stripped out so it doesn't explode the json blob.

Seen here: https://github.com/bcgov/lucy-web/pull/882/files#diff-99a086dc30892f0c40593bfccbe28751R75-R80

**Comment by NickPhura** [comment]
Hmm, one side effect of enabling the activity postgis stuff, is that the endpoint when running in docker locally fails. since the docker database doesn't have the postgis stuff installed.  Will need to see how we can address that.

Would impact testing against the PR builds too.

**Pull request merged**